### PR TITLE
Add "finder" to scripts

### PR DESCRIPTION
**Pull request by yuriramocan**
#### What does this PR do?
Adds a feature called "finder" to Hubot. It allows you to search for nearby places by making requests to the Yelp API. This version only works for the Detroit area.

#### Motivation and Context
Because sometimes I don't know what I want to eat.

#### Where should the reviewer start?
`scripts/finder.coffee` and `package.json` if you really want to make sure everything looks okay.

#### How should this be tested?
Run `hubot find me <thing>` or `hubot find me a <thing>` to search for nearby places matching your query.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### To Do:
- Add ability to search other cities/towns

### CODE REVIEW

**Comment by nwest** [comment]
You'll need to store those API keys in the .env file. see: https://github.com/detroit-labs/hubot-badger/blob/master/bin/hubot#L5

Use like this: https://github.com/detroit-labs/hubot-badger/blob/master/scripts/github-project-status.coffee#L28

Please regenerate the keys and don't check them in :)

File: scripts/finder.coffee [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    args = msg.message.text.split " "
+    terms = msg.match[1]
+
+    msg.send "Finding " + terms + " in the Detroit area:"
+
+    yelp.search { term: terms, location: 'Detroit' }
+    .then (data) ->
+      msg.send "— " + result.name + " (" + result.rating + " stars)" for result in data.businesses
+    .catch (err) ->
+      msg.send "There was an error locating your search results. Blame @nate-west-party-of-one."
```
Comment (nwest) [inline scripts/finder.coffee]: :trollface:

**Comment by yuriramocan** [comment]
@nwest Okay that's what I thought so I made a dummy account and used those keys, but can you help me understand how Hubot maintains access to those keys once I push them to Github? When I looked at the .gitignore I saw that .env was ignored, which makes sense for security, but doesn't that also prevent the variables I set from getting pushed as well?

**Comment by nwest** [comment]
Yeah we'll need to set those on heroku. I can walk you through it.

**Comment by yuriramocan** [comment]
Ohhhh this runs on Heroku? I know how to do that

**Comment by yuriramocan** [comment]
Sorry that went over my head

File: scripts/finder.coffee [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Description:
+#   Finder command for Hubot which pulls from the Yelp API.
+#   Can only search in Detroit right now.
+#
+# Dependencies:
+#   yelp
```
Comment (nwest) [inline scripts/finder.coffee]: Can you add bitly here?

**Pull request merged**